pkg/operator/baremetal_test.go

```diff
@@ -0,0 +1,98 @@
+package operator
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakekube "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestGenerateRandomPassword(t *testing.T) {
+	pwd := generateRandomPassword()
+	if pwd == "" {
+		t.Errorf("Expected a valid string but got null")
+	}
+}
+
+func newOperatorWithBaremetalConfig() *OperatorConfig {
+	return &OperatorConfig{
+		targetNamespace,
+		Controllers{
+			"docker.io/openshift/origin-aws-machine-controllers:v4.0.0",
+			"docker.io/openshift/origin-machine-api-operator:v4.0.0",
+			"docker.io/openshift/origin-machine-api-operator:v4.0.0",
+		},
+		BaremetalControllers{
+			"quay.io/openshift/origin-baremetal-operator:v4.2.0",
+			"quay.io/openshift/origin-ironic:v4.2.0",
+			"quay.io/openshift/origin-ironic-inspector:v4.2.0",
+			"quay.io/openshift/origin-ironic-ipa-downloader:v4.2.0",
+			"quay.io/openshift/origin-ironic-rhcos-downloader:v4.2.0",
+			"quay.io/openshift/origin-ironic-static-ip-manager:v4.2.0",
+		},
+	}
+}
+
+//Testing the case where the password does not already exist
+func TestCreateMariadbPasswordSecret(t *testing.T) {
+	kubeClient := fakekube.NewSimpleClientset(nil...)
+	operatorConfig := newOperatorWithBaremetalConfig()
+	client := kubeClient.CoreV1()
+
+	if err := createMariadbPasswordSecret(kubeClient.CoreV1(), operatorConfig); err != nil {
+		t.Fatalf("Failed to create new Mariadb password. %s ", err)
+		return
+	}
+	maridbPassword, err := client.Secrets(operatorConfig.TargetNamespace).Get(baremetalSecretName, metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("Error while getting newly created Mariadb password.")
+		return
+	}
+
+	password, _ := maridbPassword.StringData["password"]
+	if password == "" {
+		t.Errorf("Did not find newly created Mariadb password.")
+		return
+	}
+}
+
+//Testing the case where the password does already exist
+func TestCreateDuplicateMariadbPasswordSecret(t *testing.T) {
+	kubeClient := fakekube.NewSimpleClientset(nil...)
+	operatorConfig := newOperatorWithBaremetalConfig()
+	client := kubeClient.CoreV1()
+
+	// First create a mariadb password secret
+	if err := createMariadbPasswordSecret(kubeClient.CoreV1(), operatorConfig); err != nil {
+		t.Errorf("Failed to create new Mariadb password. %s ", err)
+		return
+	}
+	// Make sure mariadb password exists.
+	oldMaridbPassword, _ := client.Secrets(operatorConfig.TargetNamespace).Get(baremetalSecretName, metav1.GetOptions{})
+	oldPassword, _ := oldMaridbPassword.StringData["password"]
```

Comment:
this should fatal if `!ok`?

Reply:
`newPassword, ok := newMaridbPassword.StringData["password"]
if !ok {
  then fatal
}
`